src/timeline/search.rs

```diff
@@ -0,0 +1,304 @@
+
+                let allfieldinfo = match utils::get_serde_number_to_string(
+                    &record.record["Event"]["EventData"],
+                    true,
+                ) {
+                    Some(eventdata) => eventdata,
+                    _ => CompactString::new("-"),
+                };
+
+                self.search_result.insert((
+                    timestamp.into(),
+                    hostname,
+                    channel,
+                    eventid.into(),
+                    recordid,
+                    allfieldinfo,
+                    self.filepath.clone(),
+                ));
+            }
+        }
+    }
+}
+
+/// 検索結果を標準出力もしくはcsvファイルに出力する関数
+pub fn search_result_dsp_msg(
+    result_list: &HashSet<(
+        CompactString,
+        CompactString,
+        CompactString,
+        CompactString,
+        CompactString,
+        CompactString,
+        CompactString,
+    )>,
+    event_timeline_config: &EventInfoConfig,
+    output: &Option<PathBuf>,
+) {
+    let header = vec![
+        "Timestamp",
+        "Hostname",
+        "Channel",
+        "Event ID",
+        "Record ID",
+        "EventTitle",
+        "AllFieldInfo",
+        "EvtxFile",
+    ];
+    let target: Box<dyn io::Write> = match output {
+        Some(path) => match File::create(path) {
+            Ok(file) => Box::new(BufWriter::new(file)),
+            Err(err) => {
+                AlertMessage::alert(&format!("Failed to open file. {err}")).ok();
+                process::exit(1)
+            }
+        },
+        None => Box::new(BufWriter::new(io::stdout())),
+    };
+    let mut wtr = if output.is_none() {
+        Some(WriterBuilder::new().from_writer(target))
+    } else {
+        Some(WriterBuilder::new().delimiter(b',').from_writer(target))
+    };
+
+    // Write header
+    if output.is_some() {
+        wtr.as_mut().unwrap().write_record(&header).ok();
+    } else if output.is_none() && !result_list.is_empty() {
+        wtr.as_mut().unwrap().write_field(header.join(" ‖ ")).ok();
+    }
+
+    // Write contents
+    for (timestamp, hostname, channel, event_id, record_id, all_field_info, evtx_file) in
+        result_list
+    {
+        let event_title =
+            if let Some(event_info) = event_timeline_config.get_event_id(channel, event_id) {
+                event_info.evttitle.as_str()
+            } else {
+                "-"
+            };
+        let record_data = vec![
+            timestamp.as_str(),
+            hostname.as_str(),
+            channel.as_str(),
+            event_id.as_str(),
+            record_id.as_str(),
+            event_title,
+            all_field_info.as_str(),
+            evtx_file.as_str(),
+        ];
+        if wtr.as_ref().is_some() {
+            wtr.as_mut().unwrap().write_record(&record_data).ok();
+        } else {
+            wtr.as_mut()
+                .unwrap()
+                .write_field(record_data.join(" ‖ "))
+                .ok();
```

Comment:
@hitenkoku 
Omg, I just wanted to say thank you for implementing it quickly and easily, as I struggled to divide the search results with `||`.  Thank you:pray: